torch/_export/exported_program.py

```diff
@@ -164,10 +191,106 @@ def transform(self, *passes: PassType) -> "ExportedProgram":
         )
         return transformed_ep
 
-    def _add_runtime_assertions(self) -> "ExportedProgram":
-        return self.transform(
-            _AddRuntimeAssertionsForConstraintsPass(self.range_constraints, self.equality_constraints)
+    def _add_runtime_assertions(
+        self, functionalize_assertions: bool,
+    ) -> "ExportedProgram":
+        ep = self.transform(
+            _AddRuntimeAssertionsForConstraintsPass(
+                self.range_constraints,
+                self.equality_constraints,
+            ),
+        )
+        if functionalize_assertions:
+            # Ideally `_update_after_adding_runtime_assertions` should run
+            # whenever `_AddRuntimeAssertionsForConstraintsPass` runs, here
+            # bundle it with `functionalize_assertions` for safety.
+            ep = _update_after_adding_runtime_assertions(self, ep)
```

Comment:
This should also run outside of this if condition right. Since it is a bug that the signature is not updated, we should also run this before the if statement.

Reply:
Yes. As mentioned in the comment I added there just for safety but if we're OK with moving it out, let me just move it out then.